Proposal for error-handling (#2077)
dedis/cothority: .github/PULL_REQUEST_TEMPLATE.md

- [ ] 1. Design choices, user documentation and any additional doc are added (or updated) in READMEs.
- [ ] 2. Any new behaviour is tested and small units of code that can be are unit tested.
- [ ] 3. Code comments are added on tests to explain what they do.

> - [ ] 4. Errors are systematically wrapped using `xerrors` AND a meaningful message is added.

Reviewer: ```suggestion
- [ ] 4. Errors are systematically wrapped with a meaningful message using `xerrors.Errorf` and the `%+v` verb.
```
